src/nupic/utils/VectorHelpers.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef NUPIC_UTILS_VECTORHELPERS
+#define NUPIC_UTILS_VECTORHELPERS
+
+#include <vector>
+#include <string>
+#include <iostream>
+#include <sstream>
+
+#include <nupic/types/Types.hpp>
+
+
+namespace nupic {
+namespace utils {
+
+class VectorHelpers
+{
+    public:
+        /**
+        * pretty print a vector v, with separator string sep, to output stream os (default cout)
+        */
+        template<typename T>
+        static void print_vector(const std::vector<T>& v, std::string sep="", std::string prefix="", std::ostream& os=std::cout);
+        /**
+        * cast a vector to a different (compatible) type
+        */
+        template<typename T1, typename T2>
+        static std::vector<T2> castVectorType(const std::vector<T1>& orig);
+        static std::vector<Real> stringToFloatVector(const std::vector<std::string>& orig);
+        /**
+        * convert binary to sparse representation
+        */
+        template<typename T>
+        static std::vector<UInt> binaryToSparse(const std::vector<T>& binaryVector);
+        /**
+        * convert sparse to binary representation
+        */
+        template<typename T>
+        static std::vector<T> sparseToBinary(const std::vector<UInt>& sparseVector, UInt width);
+        /**
+        * convert representation of active cells(binary vector) to active columns(binary vector). 
+        * If any cell of a column is active (1), the column is considered active. See TP for details.
+        */
+        static std::vector<UInt> cellsToColumns(const std::vector<UInt>& cellsBinary, const UInt cellsPerColumn);
+};
+
+#include "nupic/utils/VectorHelpers.cpp"
```

Comment:
Templates are evaluated at compile-time so any templated types/functions have to be in the headers. Do not include `cpp` files anywhere.